File: ruby/lib/soundplayer.rb
# Created by Satoshi Nakagawa.
# You can redistribute it and/or modify it under the Ruby's license or the GPL2.

module SoundPlayer
  
  def play(name)
    return if name == nil || name.empty?
    if name == 'Beep'
      OSX.NSBeep
    else
      s = OSX::NSSound.soundNamed(name)
      s.play if s
    end
  end
  
  extend self
end

File: objc/Helpers/SoundPlayer.m
// Created by Satoshi Nakagawa.
// You can redistribute it and/or modify it under the Ruby's license or the GPL2.

#import "SoundPlayer.h"


@implementation SoundPlayer

+ (void)play:(NSString*)name
{
	if (!name || !name.length) return;
	
	if ([name isEqualToString:@"Beep"]) {
		NSBeep();
	}
	else {
		NSSound* sound = [NSSound soundNamed:name];
		if (sound) {
			[sound play];
		}
	}
}

@end
